src/sessionmanagerplugin/session/shellsession/shellsession_unix.go

```diff
@@ -67,6 +68,14 @@ func (s *ShellSession) handleKeyboardInput(log log.T) (err error) {
 	//handle double echo and disable input buffering
 	s.disableEchoAndInputBuffering()
 
+	//switch terminal in raw mode
+	oldState, err := term.MakeRaw(int(os.Stdin.Fd()))
+	if err != nil {
+		log.Errorf("Error switching terminal to raw mode: %s", err)
+		return
+	}
+	defer term.Restore(int(os.Stdin.Fd()), oldState)
```

Comment:
Auto-review: this could be moved to "Stop()" too, I don't have a strong preference. The behaviour doesn't change when I do it (we just need to add a new field `originalTermState term.State` on the `ShellSession` struct.